onnx/defs/generator/defs.cc

```diff
@@ -225,3 +225,56 @@ TensorProto message, and be valid as an output type.
         }
         propagateShapeFromInputToOutput(ctx, 0, 0);
     });
+
+ONNX_OPERATOR_SCHEMA(Multinomial)
+    .SetDoc(R"DOC(
+Generate a tensor of samples from a multinomial distribution according to the probabilities
+of each of the possible outcomes.
+)DOC")
+    .Attr(
+        "sample_size",
+        "Number of times to sample.",
+        AttributeProto::INT,
+        static_cast<int64_t>(0))
+    .Attr(
+        "seed",
+        "(Optional) Seed to the random generator, if not specified we will auto generate one.",
+        AttributeProto::FLOAT,
+        OPTIONAL)
+    .Input(
+        0,
+        "input",
+        "Input tensor with shape [batch_size, class_size], where class_size is the number of all possible outcomes.",
+        "T1")
+    .Output(
+        0,
+        "output",
+        "Output tensor with shape [batch_size, sample_size], where sample_size is the number of times to sample. ",
+        "T2")
+    .TypeConstraint(
+        "T1",
+        {"tensor(float16)", "tensor(float)", "tensor(double)"},
+        "Constrain input types to float tensors.")
+    .TypeConstraint(
+        "T2",
+        {"tensor(int32)", "tensor(int64)"},
+        "Constrain output types to integral tensors.")
+    .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+      auto output_type = ctx.getOutputType(0);
+      auto input_type = ctx.getInputType(0);
+      if (TypeProto::kTensorType != output_type->value_case() ||
```

Comment:
input_type and output_type can be `nullptr`

Reply:
add some test cases?

Reply:
Shape inference test will also be helpful here.

Reply:
This is fixed in the new revision.